main/command/src/main/scala/sbt/State.scala

```diff
@@ -182,11 +182,8 @@ object State {
         log.debug(s"> $cmd")
         f(cmd, s.copy(remainingCommands = remainingCommands, history = cmd :: s.history))
       }
-      def isInteractive = System.console != null
```

Comment:
`interactive` in State is an indication as to whether sbt is in interactive mode, i.e it's in the process of running `shell`.

This "interactive" is whether the JVM is running in interactive mode: by virtue of having a stdin and stdout attached.